### PR TITLE
feat: Add wallet automation test plan

### PR DESCRIPTION
**Pull request by SotaTek-DuyLe**
# Description

Add wallet automation test plan

### CODE REVIEW

**Review by kukkok3 — APPROVED** [review]
LGTM but wait for @dtscalac review as well before merging

File: docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md
Comment (dtscalac) [inline docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md:57]: can be ~archived~ achieved only

File: docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md
Comment (dtscalac) [inline docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md:66]: CIP-36 is something that we will aim to achieve using catalyst cardano but the library mainly targets CIP-30 and CIP-95

File: docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md
Comment (dtscalac) [inline docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md:91]: I would consider this as a test case group for the following reason: Transaction is a set of different fields, here's the whole list: https://github.com/input-output-hk/catalyst-CIPs/blob/x509-rbac-signing-with-cip30/CIP-XXXX/README.md#specification

We don't aim to support everything as some fields are optional and we don't need them but in general we should test some combinations of fields. One example is the transaction metadata. It's optional but we will need it in catalyst voices. Thus we should have at least two tests: for a transaction with, and without metadata.

Here are the fields that we support: https://github.com/input-output-hk/catalyst-voices/tree/feat/frontend/catalyst_voices_packages/catalyst_cardano_serialization#supported-transaction-body-fields.

From https://github.com/input-output-hk/catalyst-CIPs/blob/x509-rbac-signing-with-cip30/CIP-XXXX/README.md#specification you can read which fields are required. Try to come up with different combinations of required and optional fields.

**Comment by dtscalac** [comment]
> LGTM but wait for @dtscalac review as well before merging

I'm okay with the changes, just we should extend test cases a bit to cover different combinations, I described it in other comments. We can merge this PR if you agree and add missing things in a new one.

**Comment by kukkok3** [comment]
> > LGTM but wait for @dtscalac review as well before merging
> 
> I'm okay with the changes, just we should extend test cases a bit to cover different combinations, I described it in other comments. We can merge this PR if you agree and add missing things in a new one.

Ci is still busted, so we can add the additional testcases to this PR:)

**Review by kukkok3 — DISMISSED** [review]
LGTM

File: docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md
Comment (kukkok3) [inline docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md:7]: Can you link the actual issue with the testplan instead of the template one?

**Review by kukkok3 — CHANGES_REQUESTED** [review]

File: docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md
Comment (SotaTek-DuyLe) [reply docs/src/architecture/10_quality/testplans/flutter_cardano_integration.md]: done

**Review by kukkok3 — APPROVED** [review]
LGTM

**Pull request merged**
